components/support/sync15-traits/Cargo.toml

```diff
@@ -14,5 +14,5 @@ serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 log = "0.4"
 ffi-support = "0.4"
-url = "2.2"
+url = "2.1" # pinned to 2.1 for m-c, and 2.2 is incompatible, so don't allow that.
```

Comment:
Technically, this allows 2.2. And it's fine.

Reply:
I'm not sure what you mean here - maybe the comment is misleading?

The problem IIRC is that from 2.1 -> 2.2 they added `Into<String>` (or something like that, I last struck it mid last year) but also dropped the "old" way of getting a string, which I think was a mistake on their part as it broke semver. I can dig up the details if you are interested.

Reply:
doh, sorry, I do see what you mean, yeah

Reply:
So I don't think there's anything I can really do to stop these crates from using features in 2.2, so I guess we'll just discover if they do as we vendor new versions.

Reply:
You can do >=2.1,<2.2

Reply:
But then we will end up with a dupe of that crate in our mobile builds (some crates that aren't in m-c *do* need 2.2)